src/crepe/api/IMouseListener.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include "Event.h"
+#include "EventHandler.h"
+#include "EventManager.h"
+
+namespace crepe {
+
+/**
+ * \class IMouseListener
+ * \brief Interface for mouse event handling in the application.
+ */
+class IMouseListener {
+public:
+	IMouseListener();
+	/**
+     * \brief Constructs an IMouseListener with a specified channel.
+     * \param channel The channel ID for event handling.
+     */
+	IMouseListener(int channel);
+	virtual ~IMouseListener();
+	IMouseListener & operator=(const IMouseListener &) = delete;
+	IMouseListener(const IMouseListener &) = delete;
+	IMouseListener && operator=(const IMouseListener &&) = delete;
+	IMouseListener(IMouseListener &&) = delete;
+
+	/**
+     * \brief Move assignment operator (deleted).
+     */
+	IMouseListener & operator=(IMouseListener &&) = delete;
+
+	/**
+     * \brief Handles a mouse click event.
+     * \param event The mouse click event to handle.
+     * \return True if the event was handled, false otherwise.
+     */
+	virtual bool on_mouse_clicked(const MouseClickEvent & event) = 0;
```

Comment:
Ik zie nu trouwens ook dat al deze functies pure virtual zijn inplaats van virtual met een lege implementatie, is dit ook bewust? Waarom zou je een interface willen die je forceert *alle* handlers te implementeren?